backends/scg/scg.c

```diff
@@ -1,40 +1,55 @@
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include <X11/Xlib.h>
-#include <X11/Xproto.h>
-#include <X11/Xatom.h>
 #include <X11/extensions/Xrandr.h>
 
-#include <stdlib.h>
-#include <math.h>
+char *argv0 = "scg";
 
-int main(int argc, char **argv) {
-    Display *dpy = XOpenDisplay(NULL);
+#define usage() fprintf(stderr, "usage: %s GAMMA_R GAMMA_G GAMMA_B\n", argv0)
+
+#define die() do { usage(); exit(EXIT_FAILURE); } while (0)
 
-    if(dpy == NULL) {
-        return 12;    // exit code 12, if no display is found
+double parse_arg(const char *arg)
+{
+    errno = 0;
+    double gamma = strtod(arg, NULL);
+    if (errno) {
+        perror("strtod");
+        die();
     }
 
-    int screen = DefaultScreen(dpy);
-    Window root = RootWindow(dpy, screen);
-    XRRScreenResources *res = XRRGetScreenResourcesCurrent(dpy, root);
+    if (gamma < 0.0 || gamma > 1.0) {
+        fprintf(stderr, "gamma values must be in 0.0 .. 1.0\n");
+        die();
+    }
```

Comment:
This can be handled by `blugon.py`. `scg` is supposed to talk to X and nothing more.

Reply:
This is in fact rather defensively and should never trigger. My goal was to provide a more robust interface to scg, as the computational overhead and additional code complexity seems negligible.